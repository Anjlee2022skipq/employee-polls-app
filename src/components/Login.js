import React, { useState } from "react";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";
import { connect } from "react-redux";
import { handleLoginUser } from "../actions/AuthUser";
import { useNavigate } from "react-router-dom";

function Login(props) {
  const navigate = useNavigate();
  const [userName, setUserName] = useState(null);
  const handleUserName = (e) => {
    setUserName(e.target.value);
  };

  const handleLogin = (e) => {
    e.preventDefault();
    const { dispatch } = props;
    dispatch(handleLoginUser(userName));
    const path = window.location.pathname;

    if (path === "/login") return navigate("/");

    navigate(path);
  };

  return (
    <>
      <div className="header">
        <h3> Employee Polls</h3>
        <hr />
      </div>
      <div className="header-img">
        <img src="/assets/login.jfif" alt="img" />
      </div>
      <div className="login-section d-flex m-2 p-2 flex-column">
        <div className="d-flex p-2 justify-content-center">
          <DropdownButton
            id="login-user"
            title={userName === null ? "Select User" : userName}
            data-testid="dropdown-user"
          >
            {props.userIds.map((user) => (
              <Dropdown.Item
                as="button"
                className="login-user-item"
                key={user}
                value={user}
                onClick={handleUserName}
              >
                {user}
              </Dropdown.Item>
            ))}
          </DropdownButton>
        </div>
        <div className="d-flex p-2 justify-content-center">
          <button
            className="primary-btn"
            onClick={handleLogin}
            disabled={userName === null ? true : false}
          >
            Login
          </button>
        </div>
      </div>
    </>
  );
}
const mapStateToProps = ({ Users, AuthUser }) => {
  const userIds = Object.keys(Users);
  return {
    userIds,
    authenticated: !(AuthUser === null),
  };
};
export default connect(mapStateToProps)(Login);
