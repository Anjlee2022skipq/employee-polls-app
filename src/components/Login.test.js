import { render } from "@testing-library/react";
import Login from "./Login";
import React from "react";

import { Provider } from "react-redux";
import { store } from "../index";
import { BrowserRouter } from "react-router-dom";
describe("snapshot test", () => {
  it("create a snapshot folder", () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
