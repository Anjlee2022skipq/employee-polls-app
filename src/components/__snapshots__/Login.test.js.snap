// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test create a snapshot folder 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <h3>
           Employee Polls
        </h3>
        <hr />
      </div>
      <div
        class="header-img"
      >
        <img
          src="/assets/login.jfif"
        />
      </div>
      <div
        class="login-section d-flex m-2 p-2 flex-column"
      >
        <div
          class="d-flex p-2 justify-content-center"
        >
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-primary"
              id="login-user"
              type="button"
            >
              Select User
            </button>
          </div>
        </div>
        <div
          class="d-flex p-2 justify-content-center"
        >
          <button
            class="primary-btn"
            disabled=""
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <h3>
         Employee Polls
      </h3>
      <hr />
    </div>
    <div
      class="header-img"
    >
      <img
        src="/assets/login.jfif"
      />
    </div>
    <div
      class="login-section d-flex m-2 p-2 flex-column"
    >
      <div
        class="d-flex p-2 justify-content-center"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="login-user"
            type="button"
          >
            Select User
          </button>
        </div>
      </div>
      <div
        class="d-flex p-2 justify-content-center"
      >
        <button
          class="primary-btn"
          disabled=""
        >
          Login
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
